src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,289 @@
+            TaskArray.get(TaskArray.size() - 1).markAsDone();
+        }
+    }
+
+    private static void addEventFromFile(String fileInput) {
+        int byIndex = fileInput.indexOf("(at:");
+        Event eventAdded = new Event(fileInput.substring(7, byIndex), fileInput.substring(byIndex + 5, fileInput.indexOf(")")));
+        TaskArray.add(eventAdded);
+        if (fileInput.contains("X")) {
+            TaskArray.get(TaskArray.size() - 1).markAsDone();
+        }
+    }
+
+    private static void addTodoFromFile(String fileInput) {
+        Todo todoAdded = new Todo(fileInput.substring(7));
+        TaskArray.add(todoAdded);
+        if (fileInput.contains("X")) {
+            TaskArray.get(TaskArray.size() - 1).markAsDone();
+        }
+    }
+
+    private static void updateFile(Path filePath) throws IOException {
+        FileWriter fw = new FileWriter(String.valueOf(filePath));
+        for (Task t : TaskArray) {
+            fw.write(t.toString() + System.lineSeparator());
+        }
+        fw.write("There are a total of " + TaskArray.size() + " tasks in the list.");
+        fw.close();
+    }
+
+    public static String getUserInput(Scanner scan) {
+        String userInput;
+        userInput = scan.nextLine();
+        try {
+            handleErrorUserInputs(userInput);
+        } catch (TaskException e) {
+            invalidInput = true;
+            System.out.println("☹️ OOPS!!! The description of a todo cannot be empty.");
+            System.out.println(SECTION_DIVIDER);
+        } catch (EventException e) {
+            invalidInput = true;
+            System.out.println("☹️ OOPS!!! The description of an event cannot be empty or is incomplete (/at).");
+            System.out.println(SECTION_DIVIDER);
+        } catch (DeadlineException e) {
+            invalidInput = true;
+            System.out.println("☹️ OOPS!!! The description of a deadline cannot be empty or is incomplete (/by).");
+            System.out.println(SECTION_DIVIDER);
+        }
+        return userInput;
+    }
+
+    public static void addTaskToArray(String userInput) {
+        Task taskAdded = null;
+        if (userInput.contains("deadline")) {
+            taskAdded = addDeadlineToArray(userInput);
+        } else if (userInput.contains("event")) {
+            taskAdded = addEventToArray(userInput);
+        } else if (userInput.contains("todo")){
+            taskAdded = addTodoToArray(userInput);
+        }
+        System.out.println("Got it. I've added this task:" + System.lineSeparator() + taskAdded);
+        System.out.println("Now you have " + TaskArray.size() + " tasks in the list.");
+        System.out.println(SECTION_DIVIDER);
+    }
+
+    public static Todo addTodoToArray(String userInput) {
+        int descriptionIndex = userInput.indexOf("todo") + INDEX_TODO;
+        Todo todoAdded = new Todo(userInput.substring(descriptionIndex));
+        TaskArray.add(todoAdded);
+        return todoAdded;
```

Comment:
You could consider if there are ways of naming a [collection](https://se-education.org/guides/conventions/java/basic.html#naming) and whether verbs should be used as part of a variable name (Hint: what is todoAdded v.s. a todoItem?).